### ISSUE TITLE
Child animation clips in 2D Directional Mixer are not fully active

### ISSUE THREAD

**Issue by neo-mashiro** [KybernetikGames/animancer#96]
# Environment

- Animancer Version Number: v6.0
- Animancer Pro or Lite: Lite
- Unity Version: 2019.4.15f1
- Platform: Windows 10

# Description

I'm using a 2D directional mixer to blend animations when my player is flying. There are four simple animation clips that are controlled by thresholds and parameters as follows:

- Fly Down (X: 0, Y: -5)
- Fly Up (X: 0, Y: 5)
- Fly Idle (X: 0, Y: 0)
- Fly Move (X: 10, Y: 0)

where parameter X is the horizontal forward speed and Y is the vertical speed.

I'm handling turning and rotations manually in my character controller script so there's no "Fly Move Backward" clip.
X is always 0 or positive but Y could be negative as well.

While idle and move animations work just fine, the vertical up and down animations are often not fully played, sometimes halfway blended, sometimes jerky snaps between halfway and fully played, even if the parameters are exactly (0, 5) or (0, -5).

# Workaround

Changing the mixer type from __Directional__ to __Cartesian__ solved my problem, everything works flawlessly.
Since I'm new to Unity, I'm not sure if this is a bug or intended, just curious about why the __Directional__ algorithm failed to blend my "Fly Down", "Fly Idle", "Fly Up" clips well despite they are indeed "directional". You can have a look at this if it's not expected.

# Reproduction

I've modified the 3DGK example so I don't know how to provide an easily reproducible version, but here's my setups and the fly motion state code:
```C#
using UnityEngine;
using Animancer;

namespace Players {

    public class FlyState : MotionState {

        [SerializeField] private MixerState.Transition2D mixerTransition2D;
        [SerializeField] private float blendSpeed = 24f;

        public override bool CanExitState {
            get {
                var nextState = AnimatorController.NextState;
                if (nextState.Priority > Priority) {
                    return true;
                }

                return nextState == AnimatorController.Airborne || nextState == AnimatorController.Climb;
            }
        }
        
        private void OnEnable() {
            Debug.Log("Enter fly state");
            AnimatorController.Animancer.Play(mixerTransition2D);
        }
        
        private void OnDisable() {
            Debug.Log("Exit fly state");
        }

        private void FixedUpdate() {
            var parameters = new Vector2(AnimatorController.NextStateParameterX, AnimatorController.NextStateParameterY);
            mixerTransition2D.State.Parameter = Vector2.Lerp(
                mixerTransition2D.State.Parameter, parameters, blendSpeed * Time.deltaTime);
            Debug.Log($"{parameters}, {mixerTransition2D.State.Parameter}");
        }
    }
}

```
![Capture](https://user-images.githubusercontent.com/29720830/105383464-25815680-5bdf-11eb-974d-3faac323e822.PNG)

**Comment by neo-mashiro** [comment]
Oops, should have unchecked the "Sync" checkbox of the "Fly Idle" clip as well, but that has no effect on this issue.

**Comment by KybernetikGames** [comment]
In areas with 180 degrees or more between states the interpolation of a Directional Mixer is ill-defined and will likely have undesirable results; I.E. if you have clips for forward, back, and right, but no left, then you will get odd results if you set the parameter to the left.

You said you aren't setting the parameter to the left so I'm not sure if that's actually the problem, but that's the only thing I can think of that would really cause such a difference between Directional and Cartesian.

If you want to send a minimal reproduction project to animancer@kybernetik.com.au, I'd be happy to take a look at it.

**Comment by neo-mashiro** [comment]
@KybernetikGames Thank you for your hint, after 4 hours of trial and error, I found that's exactly where the problem is.

I previously said that X is always 0 or positive, which is not really true. Due to floating point inaccuracy, in some frames X can become a very small negative number such as -1e-10, as a result the up and down clips each has a weight of 0.5, this leads to a weird jerky blend.

Now I have changed the mixer type from __Cartesian__ back to __Directional__, and by clamping the parameter X above 0, blending looks very good.

I will close this since it's not a bug.
